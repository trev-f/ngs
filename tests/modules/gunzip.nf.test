nextflow_process {

    name "Test Process GUNZIP"
    script "modules/gunzip.nf"
    process "GUNZIP"

    test("GUNZIP successfully unzips GZIP-compressed fasta file.") {

        when {
            params {}
            process {
                """
                input[0] = file("https://github.com/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa.gz")
                """
            }
        }

        then {
            assert process.success

            with(process.out.gunzip) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I.fa/
                assert path(get(0)).md5 == "e050100d0a9a7d06a8accf8424ad0d98"
            }

            assert snapshot(process.out).match()
        }

    }

}
