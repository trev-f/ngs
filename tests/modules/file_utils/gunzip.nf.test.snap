{
    "GUNZIP successfully unzips GZIP-compressed fasta file.": {
        "content": [
            {
                "0": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ],
                "gunzip": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ]
            }
        ],
        "timestamp": "2023-04-19T18:42:42+0000"
    }
}